core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -676,6 +682,78 @@ public int attachPartition(long timestamp) {
         return StatusCode.OK;
     }
 
+    public StatusCode detachPartition(long timestamp) {
+        // Partitioned table must have a timestamp
+        assert metadata.getTimestampIndex() > -1;
+
+        try {
+            StatusCode statusCode = detachOperationOnPartition(timestamp, false);
+            if (StatusCode.OK != statusCode) {
+                return statusCode;
+            }
+
+            setPathForPartition(path, partitionBy, timestamp, false);
+            if (!ff.exists(path.$())) {
+                LOG.error()
+                        .$("partition folder does not exist [path=")
+                        .$(path)
+                        .$(']')
+                        .$();
+                return StatusCode.PARTITION_FOLDER_DOES_NOT_EXIST;
+            }
+
+            if (inTransaction()) {
+                LOG.info()
+                        .$("committing open transaction before applying detach partition command [table=")
+                        .$(tableName)
+                        .$(", partition=")
+                        .$ts(timestamp)
+                        .I$();
+                commit();
```

Comment:
I'd expect it to happen before `detachOperationOnPartition()` call. Can you add test where you add data to the partition you detach without commit and then detach it?